cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_ctrl_packages.go

```diff
@@ -127,3 +127,46 @@ func (s *Server) GetAvailablePackageSummaries(ctx context.Context, request *core
 	}
 	return response, nil
 }
+
+// GetAvailablePackageVersions returns the package versions managed by the 'kapp_controller' plugin
+func (s *Server) GetAvailablePackageVersions(ctx context.Context, request *corev1.GetAvailablePackageVersionsRequest) (*corev1.GetAvailablePackageVersionsResponse, error) {
+	log.Infof("+kapp-controller GetAvailablePackageVersions")
+
+	// Retrieve the proper parameters from the request
+	namespace := request.GetAvailablePackageRef().GetContext().GetNamespace()
+	cluster := request.GetAvailablePackageRef().GetContext().GetCluster()
+	identifier := request.GetAvailablePackageRef().GetIdentifier()
+
+	// Validate the request
+	if namespace == "" || identifier == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "Required context or identifier not provided")
+	}
+
+	if cluster == "" {
+		cluster = s.globalPackagingCluster
```

Comment:
As per the previous PR, I see no reason (yet) to not assume invalid argument if the cluster is not set.

Reply:
See https://github.com/kubeapps/kubeapps/pull/3784#discussion_r755995552